File: 1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	map1 := map[int]int{}
	for i, x := range nums {
		if p, ok := map1[target-x]; ok {
			return []int{p, i}
		}
		map1[x] = i
	}
	return nil
}

// @lc code=end

File: 1.两数之和.py
from typing import *
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i, x in enumerate(nums):
            less = target - x

            if less in map.keys():
                return [i, map[less]]
            map[x] = i

# @lc code=end

test = Solution()
print(test.twoSum([123,321,32], 155))